lib/Sema/CSApply.cpp

```diff
@@ -6501,7 +6545,8 @@ Expr *ExprRewriter::coerceToType(Expr *expr, Type toType,
   }
 
   // Unresolved types come up in diagnostics for lvalue and inout types.
-  if (fromType->hasUnresolvedType() || toType->hasUnresolvedType())
+  if (fromType->hasUnresolvedType() || toType->hasUnresolvedType() /* ||
+      fromType->is<ErrorType>()*/)
```

Comment:
Did you intend to leave this commented out?

Reply:
No, will remove.  Thanks for the catch.